lib/go-tc/users.go

```diff
@@ -136,9 +406,49 @@ type CurrentUserUpdateRequestUser struct {
 	Username           json.RawMessage `json:"username"`
 }
 
+func (u UserCurrentV40) UpgradeUserCurrent() UserCurrentV50 {
```

Comment:
GoDoc?

Also, nit: Since this can be only be called from a `UserCurrentV40`, we already know it's upgrading "from a UserCurrentV40", those two words make the name longer without giving us any extra information. Could just be `Upgrade`.